### PR TITLE
translate(yf) multi_agent_cooperation_rl_eng.rst (from #143)

### PR DESCRIPTION
**Pull request by VaninaY**
translate multi_agent_cooperation_rl_zh.rst in pr#143 to multi_agent_cooperation_rl_eng.rst

### CODE REVIEW

File: source/02_algo/multi_agent_cooperation_rl_eng.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  3. Individual goal consistency: The goal of each agent may be the optimal global return or the optimal local return.
+
+  4. Scalability: Large-scale multi-agent systems may involve high-dimensional state space and action space, which poses certain challenges for model expression ability, algorithm learning ability and hardware computing power in real scenarios.
+
+
+
+Research Direction
+-----------------------
+For the MARL cooperation task, the simplest idea is to directly apply the single-agent reinforcement learning method to the multi-agent system, that is, each agent regards other agents as factors in the environment, and still learns according to the single agent way to update the policy by interacting with the environment; This is the idea of independent Q-learning, independent PPO methods, but due to the non-stationarity of the environment and the locality of agent observations, these methods are difficult to achieve good results.
+
```
Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:41]: For MARL cooperation tasks, the simplest idea is to directly apply single-agent reinforcement learning methods to multi-agent systems. Specifically, each agent regards other agents as part of the environment, and then utilize the single agent reinforcement learning paradigms. This is the basic idea of independent Q-learning, independent PPO, but due to the non-stationarity of the environment and the locality of agent observation, these methods are difficult to achieve good results.

File: source/02_algo/multi_agent_cooperation_rl_eng.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+At present, MARL cooperation mainly adopts the method of CTDE (centralized training and decentralized execute), and there are two main types of solutions, Valued-based MARL and Actor-Critic MARL. For details, please refer to the following figure:
```
Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:42]: At present, the cooperative tasks in MARL mainly adopt the CTDE paradigm (centralized training and decentralized execution), which mainly include two types of learning methods, Valued-based MARL and Actor-Critic MARL. Please refer to Figure 2 for details:

File: source/02_algo/multi_agent_cooperation_rl_eng.rst [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+.. image:: images/MARL_cooperation_algo.png
+   :align: center
+   :scale: 50 %
+
+**Valued-based MARL**
+
+For Valued-based MARL, the main idea is to decompose the global reward value into local reward values that can be learned by each agent, so as to facilitate the training of the agent. There are mainly QMIX, WQMIX, QTRAN and other methods:
```
Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:50]: For Valued-based MARL, the main idea is the factorization of the joint action-value function :math:`Q_{tot}` into individual ones  :math:`Q_a` for decentralized execution. In order to achieve CTDE, we need to ensure that factorization satisfy the IGM assumption: the  a global argmax performed on :math:`Q_{tot}` yields the same result as a set of individual argmax operations performed on each :math:`Q_a`. *

File: source/02_algo/multi_agent_cooperation_rl_eng.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+- WQMIX: The core of WQMIX is the same as that of QMIX, and it also learns a Q-value mixed network, but it learns a Q-value mixed network that can break through the monotonicity limit through the weighted projection method. For details, please refer to `WQMIX <https://github.com/opendilab/DI-engine-docs/blob/main/source/hands_on/wqmix.rst>`_ [1]_
+
+- QTRAN: QTRAN breaks through the monotonicity limitation by learning independent action-value networks, hybrid action-value networks, and global state-value networks. For details, please refer to `QTRAN <https://github.com/opendilab/DI-engine-docs/blob/main/source/hands_on/qtran.rst>`_ [4]_
+
+- QPLEX: QPLEX decomposes the joint Q-value :math:`Q_tot` and the Q-value of each agent :math:`Q_i` using a Dueling structure, respectively. The IGM consistency is transformed into an easy-to-implement advantage function value range constraint, which facilitates the learning of value functions with a linear decomposition structure. For details, please refer to `QPLEX <https://arxiv.org/abs/2008.01062>`_ [10]_
+
+**Actor-critic MARL**
```
Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:60]: Actor-critic -> Actor-Critic

File: source/02_algo/multi_agent_cooperation_rl_eng.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+For Actor-critic MARL, the main idea is to learn a multi-agent policy network. There are mainly COMA, MAPPO and other methods:
```
Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:62]: For Actor-Critic MARL, the main idea is to use the policy gradient theorem to update the policy networks while learning a fully centralized state-action value function and use it to guide the optimization of decentralized policies.

File: source/02_algo/multi_agent_cooperation_rl_eng.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+.. image:: images/MARL_cooperation_algo.png
+   :align: center
+   :scale: 50 %
+
+**Valued-based MARL**
+
+For Valued-based MARL, the main idea is to decompose the global reward value into local reward values that can be learned by each agent, so as to facilitate the training of the agent. There are mainly QMIX, WQMIX, QTRAN and other methods:
+
+- QMIX: The core of QMIX is to learn a monotonic Q-value mixed network, and the Q-value of each agent is summed by nonlinear transformation to generate :math:`Q_tot`. For details, please refer to `QMIX <https://github.com/opendilab/DI-engine-docs/blob/main/source/hands_on/qmix.rst>`_ [2]_
+
```
Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:53]: - 全部需要修改到新的文档链接 https://github.com/opendilab/DI-engine-docs/blob/main/source/12_policies/qmix.rst
- 中文文档中也是

Comment (puyuan1996) [inline source/02_algo/multi_agent_cooperation_rl_eng.rst:53]: 这里各个算法的中文文档介绍，叫昱宏再修改扩充一下，然后再翻译为英文，现在的不太清晰

Comment (VaninaY) [reply source/02_algo/multi_agent_cooperation_rl_eng.rst:53]: All done except MAPPO. But the hyperlink doesn't work for an unknown reason: { when you click on it, the message says: "The 'opendilab/DI-engine-docs' repository doesn't contain the 'source/hands_on/qmix.rst' path in 'main'."}

**Pull request merged**